api/teamserver/rename.go

```diff
@@ -0,0 +1,74 @@
+package teamserver
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/auth"
+)
+
+// RenameTeam allows an authed user with authority or admin to rename a team
+func (s *Server) RenameTeam(w http.ResponseWriter, r *http.Request) {
+	logger := s.logger.Session("rename-team")
+
+	authTeam, authTeamFound := auth.GetTeam(r)
+	if !authTeamFound {
+		logger.Error("failed-to-get-team-from-auth", errors.New("failed-to-get-team-from-auth"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	teamName := r.FormValue(":team_name")
+	if !authTeam.IsAdmin() && !authTeam.IsAuthorized(teamName) {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logger.Error("failed-to-read-body", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var rename atc.RenameRequest
+	err = json.Unmarshal(data, &rename)
+	if err != nil {
+		logger.Error("failed-to-unmarshal-body", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	team, found, err := s.teamFactory.FindTeam(teamName)
+	if err != nil {
+		logger.Error("failed-to-get-team", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if !found {
+		logger.Info("team-not-found")
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	fmt.Println(team.Name(), atc.DefaultTeamName)
+	if team.Name() == atc.DefaultTeamName {
```

Comment:
is this necessary? renaming the 'main' team should be fine - all logic is based on it being 'admin: true', not by name

Reply:
At first my thought was we shouldn't rename it but you are right, I am not sure there is anything really preventing us from allowing it. I will remove it.